eng/source-build-pre-release.yml

```diff
@@ -67,9 +71,37 @@ jobs:
       runId: $(InstallerOfficialRunId)
 
   - template: templates/steps/upload-to-blob-storage.yml
+    condition: and(succeeded(), ne('${{ parameters.isDryRun }}', 'true'))
     parameters:
       file: '$(PIPELINE.WORKSPACE)/dotnet-sdk-source-$(SdkVersion).tar.gz'
       accountName: '$(storageAccountName)'
       containerName: '$(blobContainerName)'
       uploadPath: '$(blobContainerUploadBaseFilePath)/$(ReleaseChannel)/$(RuntimeVersion)-$(SdkVersion)'
       azureStorageKey: '$(dotnetclimsrc-access-key)'
+  
+  - script: |
+      set -euo pipefail
+      
+      upstream_with_pat=$(echo $(vmrUpstreamUrl) | sed "s,https://.*@,https://dn-bot:${AZDO_PAT}@,g")
+
+      $(Build.SourcesDirectory)/eng/update-vmr.sh \
+        --releaseChannel $(ReleaseChannel) \
+        --sdkVersion $(SdkVersion) \
+        --upstream ${upstream_with_pat} \
+        --tarball $(Pipeline.Workspace)/dotnet-sdk-source-$(SdkVersion).tar.gz
+
+      month_year=$(date +"%b%Y" -d "+1 month" | sed 's/.*/\L&/') # e.g. aug2022
```

Comment:
There seems to be some coupling and duplication between this pipeline logic and the script.  e.g. the branch name algorithm is duplicated and the dotnet-vmr path is hard coded across the two.  These can cause maintenance issues in the future if they ever need to be changed.  Consider pushing all of the logic into the script and make this step a simple invocation of the script.  This would solve the coupling as well as make it easier to test locally.  If beneficial for local testing purposes, the push logic could be optional controlled by a parameter.